src/CRUD.java

```diff
@@ -0,0 +1,48 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class CRUD<T extends Base> {
+
+	    List<T> list = null;
+	    
+	    void read(){
+	        if(!list.isEmpty()) {
+	        	for (T t : list) {
+					System.out.println(t.toString() + "\n");
+				}
+	        }
+	        else {
+	        	throw new NullPointerException("The list is empty.");
+	        }
+	    }
+	    
+	    List<T> getList(){
+	    	return list;
+	    }
+	    
+	    void initialize() {
+	    	list= new ArrayList<T>();
+	    }
+	    
+	    void delete(int id) {
+	    	T obj=find(id);
+	    	list.remove(obj);
+	    }
+	    
+	    T find(int id) {
+	    	T obj=null;
+	    	if(!list.isEmpty()) {
```

Comment:
I think you need to catch the Exception with a try/catch like...
```
try {
    T obj = null
    if (!list.isEmpty()) {
        for (T t : list) {
            if (t.getId() == id; {
                obj = t;
            }
        }
    }
} catch (NullPointerException e) {
    // code
} catch (Exception e) {
    // code
}

return result;
```

But we can discuss this with @AlejandroWoloch